pkg/logproto/logproto.proto

```diff
@@ -311,8 +307,8 @@ message GetChunkRefRequest {
     (gogoproto.nullable) = false
   ];
   string matchers = 3;
-  repeated LineFilter filters = 4 [
-    (gogoproto.customtype) = "github.com/grafana/loki/pkg/logql/syntax.LineFilter",
+  Plan plan = 4 [
+    (gogoproto.customtype) = "github.com/grafana/loki/pkg/querier/plan.QueryPlan",
```

Comment:
Changing the type of a field is a breaking/incompatible change (between weekly releases). We should avoid that.

Reply:
Just add the plan as field index 5 and remove field 4 the after rollout

Reply:
Good call! I will do that :)

Reply:
Done